### DISCUSSION TITLE
Is daisyui compatible with talwindcss 3 ?

### DISCUSSION THREAD

**Discussion by pooyaEst** [saadeghi/daisyui#334] [Q&A] [answered]
It has been released recently, should we wait before upgrading ?

**Comment by saadeghi** [comment, marked as answer]
@pooyaEst It is compatible 👍  
I'm using version `3.0` myself and there are no issues.

**Reply by mainick** [reply]
@saadeghi can you share your configuration?

**Reply by saadeghi** [reply]
Configuration of what?  
You can just install the latest version of everything on any environments and start using them:
```
npm i tailwindcss@latest daisyui@latest
```

**Reply by mainick** [reply]
Here is my package.json
`{
  "name": "learning",
  "version": "1.0.0",
  "private": true,
  "dependencies": {
    "@testing-library/jest-dom": "^5.11.4",
    "@testing-library/react": "^11.1.0",
    "@testing-library/user-event": "^13.1.8",
    "axios": "^0.24.0",
    "daisyui": "^1.16.5",
    "lodash": "^4.17.21",
    "prop-types": "^15.7.2",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-error-boundary": "^3.1.3",
    "react-query": "^3.33.4",
    "react-router-dom": "^5.2.0",
    "react-scripts": "4.0.3",
    "react-toastify": "^8.1.0",
    "recoil": "^0.4.1",
    "sweetalert2": "^11.3.0",
    "sweetalert2-react-content": "^4.2.0",
    "uuid": "^8.3.2",
    "web-vitals": "^1.0.1"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject",
    "prepare": "husky install",
    "format": "prettier --write .",
    "lint": "eslint . --ext ts --ext tsx",
    "clean": "rm -rf ./node_modules"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "autoprefixer": "^10.4.0",
    "eslint": "^7.25.0",
    "eslint-config-airbnb": "^18.2.1",
    "eslint-config-prettier": "^8.3.0",
    "eslint-plugin-import": "^2.22.1",
    "eslint-plugin-jest": "^24.3.6",
    "eslint-plugin-jsx-a11y": "^6.4.1",
    "eslint-plugin-prettier": "^3.4.0",
    "eslint-plugin-react": "^7.23.2",
    "eslint-plugin-react-hooks": "^4.2.0",
    "husky": ">=6",
    "json-server": "^0.17.0",
    "lint-staged": ">=10",
    "postcss": "^8.4.5",
    "prettier": "^2.2.1",
    "tailwindcss": "^3.0.5"
  },
  "lint-staged": {
    "*.{js,jsx}": [
      "npm lint",
      "npm format"
    ]
  }
}
`
I no longer use craco to start the application.
This is the tailwindcss configuration
`module.exports = {
  content: ['./src/**/*.{js,jsx,ts,tsx}', './public/**/*.html'],
  theme: {
    extend: {},
  },
  // eslint-disable-next-line global-require
  plugins: [require('daisyui')],
  daisyui: {
    styled: true,
    themes: ['cupcake', 'dark'],
    base: true,
    utils: true,
    logs: true,
    rtl: false,
  },
}
`

**Reply by saadeghi** [reply]
I'm still not getting what's the problem here.  
If you're using Tailwind CSS and daisyUI but there are issues or unexpected behaviors, please open a new issue and provide an  example (a small GitHub repo) so I can reproduce the issue and help you fix it.

**Reply by Benoit1980** [reply]
After reading this post, I realised that you are just adding a layer on the top of Tailwind css but in no case are stuck to a specific version of Tailwind to keep going.
I think people believe that your library must work in parallel with the newest Tailwind updates. If you think about it, when most of us download or try a product, we do not always understand at first how things are put together behind the scenes. So I think @pooyaEst 
 was genuinely asking if your version 1.16.6 would still work the same way with Tailwind 2 and Tailwind 3. That's all.

**Reply by pooyaEst** [reply]
I thought there might be few changes in some class names and because daisyui applies tailwindcss class names it might turn into a problem in some components which fortunatly isnt the case

**Reply by Benoit1980** [reply]
yes and I understand what you mean because I thought of the same :-)
